packages/next-intl/src/middleware/utils.tsx

```diff
@@ -19,10 +87,13 @@ export function getInternalTemplate<
   pathname: string,
   locale: Locales[number]
 ): [Locales[number] | undefined, keyof Pathnames | undefined] {
+  // Sort pathnames by specificity
+  const sortedPathnames = getSortedPathnames(pathnames);
```

Comment:
Would suggest to add a line break before the next comment just to bikeshed a bit about code aesthetics 😄 

```suggestion
  const sortedPathnames = getSortedPathnames(pathnames);

```